Fitness/Controllers/HomeController.swift

```diff
@@ -89,61 +94,83 @@ class HomeController: UIViewController {
         statusBarBackgroundColor = UIView(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 21))
         statusBarBackgroundColor.backgroundColor = .fitnessWhite
         view.addSubview(statusBarBackgroundColor)
+        
+        // GET HABITS
+        habits = Habit.getActiveHabits()
+        mainCollectionView.reloadSections(IndexSet(integer: 0))
 
         // GET GYMS
-        NetworkManager.shared.getGyms { (gyms) in
+        NetworkManager.shared.getGyms { gyms in
             self.gyms = gyms.sorted { $0.isOpen && !$1.isOpen }
-            self.mainCollectionView.reloadSections(IndexSet(integer: 0))
+            
+            let gymNames = UserDefaults.standard.stringArray(forKey: Identifiers.favoriteGyms) ?? []
+            self.updateFavorites(favorites: gymNames)
+            
+            self.mainCollectionView.reloadSections(IndexSet(integer: 1))
         }
 
         // GET TODAY'S CLASSES
-
         let stringDate = Date.getNowString()
-        print("TRACE: today: \(stringDate)")
 
         NetworkManager.shared.getGymClassesForDate(date: stringDate) { (gymClassInstances) in
             self.gymClassInstances = gymClassInstances.sorted { (first, second) in
                 return first.startTime < second.startTime
             }
-            self.mainCollectionView.reloadSections(IndexSet(integer: 1))
-        }
-
-        // GET TAGS
-        NetworkManager.shared.getTags { tags in
-            self.tags = tags
             self.mainCollectionView.reloadSections(IndexSet(integer: 2))
         }
+
     }
     
     override func viewDidAppear(_ animated: Bool) {
-        for cell in mainCollectionView.visibleCells {
+        super.viewDidAppear(animated)
+        
+        for cell in allGymsCollectionView?.visibleCells ?? [] {
             if let cell = cell as? GymsCell, let indexPath = mainCollectionView.indexPath(for: cell) {
-                cell.setGymStatus(isOpen: gyms[indexPath.item].isOpen)
+                cell.setGymStatus(isOpen: favoriteGyms[indexPath.item].isOpen, changingSoon: isGymStatusChangingSoon(gym: favoriteGyms[indexPath.item]))
             }
         }
     }
-    
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.tabBarController?.tabBar.isHidden = false
+        
+        let newHabits = Habit.getActiveHabits()
+        if habits != newHabits {
+            habits = newHabits
+            mainCollectionView.reloadSections(IndexSet(integer: 0))
+        }
+    }
+
     func getCollectionViewCellSize(_ collectionView: UICollectionView, identifier: String) -> CGSize {
-        let allGymsCellSize = CGSize(width: (collectionView.bounds.width - 32.0 - 12.0)/2.0, height: 60.0)
+        // Habits
+        let checkInCellSize = CGSize(width: collectionView.bounds.width, height: 53.0)
```

Comment:
Constants! Bring all the heights out to top of the function

Reply:
sounds good!